Added recipe for language-detection

andreasjansson opened the pull request:
### Brief summary of what the package does

Automatic programming language detection from snippets of code.

### Direct link to the package repository

https://github.com/andreasjansson/language-detection.el

### Your association with the package

I'm the maintainer.

### Relevant communications with the upstream package maintainer

None needed

### Checklist

- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] I've built and installed the package using the instructions in the [README](https://github.com/melpa/melpa/blob/master/README.md)

syohex commented on recipes/language-detection:
```diff
@@ -0,0 +1,3 @@
+(language-detection :fetcher github
+                    :repo "andreasjansson/language-detection.el"
+                    :files ("language-detection.el"))
```
You need not to specify `:files` attribute, it is enough to use default value.

Please read 
- https://github.com/melpa/melpa#recipe-format

syohex commented:
- No autoload cookie
- No `provide`

andreasjansson commented:
Thanks for review @syohex. Addressed comments in https://github.com/andreasjansson/language-detection.el/commit/3b180a5a0416286051e39e511ce51400c21310f0

Fanael commented:
You declare a dependency on `cl-lib` but don't use it. Could you change `loop` to `cl-loop` and let the old `cl` library rest in peace?

andreasjansson commented:
Thanks @Fanael, I changed the [`(require 'cl)` to `(require 'cl-lib)`](https://github.com/andreasjansson/language-detection.el/blob/master/language-detection.el#L72) and [`loop` to `cl-loop`](https://github.com/andreasjansson/language-detection.el/blob/master/language-detection.el#L96). I also [removed the dependency on cl-lib](https://github.com/andreasjansson/language-detection.el/blob/master/language-detection.el#L7) since I assume that's bundled with Emacs and doesn't need an explicit require. Is that correct?

Fanael commented:
It's only bundled with Emacs 24.3 and newer. Either declare a dependency on that, or put the `cl-lib` dependency back in.

andreasjansson commented:
I put the [`cl-lib` dependency back in](https://github.com/andreasjansson/language-detection.el/blob/master/language-detection.el#L7).

purcell commented:
Thanks. I think `language-detection-buffer` would also be useful programmatically, so perhaps change the body of it so that it always returns the detected language, and only prints it when called interactively.
